utility_analysis_new/dp_engine.py

```diff
@@ -33,53 +33,42 @@ def __init__(self, budget_accountant: budget_accounting.BudgetAccountant,
                  backend: pipeline_backend.PipelineBackend):
         super().__init__(budget_accountant, backend)
         self._is_public_partitions = None
-        self._sampling_probability = 1.0
 
-    def aggregate(self,
-                  col,
-                  params: pipeline_dp.AggregateParams,
-                  data_extractors: pipeline_dp.DataExtractors,
-                  public_partitions=None,
-                  multi_param_configuration: Optional[
-                      utility_analysis_new.MultiParameterConfiguration] = None,
-                  partitions_sampling_prob: float = 1.0):
+    def analyze(self,
+                col,
+                options: utility_analysis_new.UtilityAnalysisOptions,
+                data_extractors: pipeline_dp.DataExtractors,
+                public_partitions=None):
         """Performs utility analysis for DP aggregations per partition.
 
         Args:
           col: collection where all elements are of the same type.
-          params: specifies which metrics to compute and computation parameters.
+          options: options for utility analysis.
           data_extractors: functions that extract needed pieces of information
             from elements of 'col'.
           public_partitions: A collection of partition keys that will be present
             in the result. If not provided, the utility analysis with private
             partition selection will be performed.
-          multi_param_configuration: if provided the utility analysis for
-            multiple parameters will be performed: 'params' is used as
-            blue-print and non-None attributes from 'multi_param_configuration'
-            are used for creating multiple AggregateParams. See docstring for
-            MultiParameterConfiguration for more details.
-          partitions_sampling_prob: todo
 
         Returns:
             A collection with elements (pk, utility analysis metrics).
```

Comment:
nit: I would write "partition_key" so that it's not confused with "privacy_key".

Reply:
Thanks! For docstrings of course we should use the whole name